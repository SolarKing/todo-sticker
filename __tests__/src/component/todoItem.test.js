// @flow
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import TodoItem from '../../../src/component/todoItem';

describe('TodoItem', () => {
    let wrapper: ShallowWrapper;

    beforeAll(() => {
        wrapper = shallow(
            <TodoItem
                todo={{
                    id: '1',
                    dueDatetime: 10,
                    description: 'Hello!',
                    complete: false,
                    createTime: 0,
                }}
            />
        );
    });

    it('should render correct with short description', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct with long description', () => {
        wrapper.setProps({
            todo: {
                id: '1',
                dueDatetime: 10,
                description: 'Hello!\rLineTwo\rLineThree',
                complete: false,
                createTime: 0,
            }
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct with completed', () => {
        wrapper.setProps({
            todo: {
                id: '1',
                dueDatetime: 10,
                description: 'Hello!',
                complete: true,
                createTime: 0,
            }
        });
        expect(wrapper).toMatchSnapshot();
    });

});