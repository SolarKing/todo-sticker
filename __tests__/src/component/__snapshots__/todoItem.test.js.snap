// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render correct with completed 1`] = `
<TodoItemDiv>
  <todoItem__LowDiv>
    <todoItem__ClockDiv>
      <Icon
        type="clock-circle-o"
      />
    </todoItem__ClockDiv>
    <div
      style={
        Object {
          "gridArea": "notification-time",
        }
      }
    >
      1970/01/01 08:00
    </div>
    <todoItem__CreatedTimeDiv>
      <span>
        Creat at: 
      </span>
      1970/01/01 08:00
    </todoItem__CreatedTimeDiv>
    <todoItem__DescriptionDiv>
      <p>
        Hello!
      </p>
    </todoItem__DescriptionDiv>
    <todoItem__ActionButtonDiv>
      <Connect(TodoActionButton)
        todo={
          Object {
            "complete": true,
            "createTime": 0,
            "description": "Hello!",
            "dueDatetime": 10,
            "id": "1",
          }
        }
      />
    </todoItem__ActionButtonDiv>
    <todoItem__CompleteMarkDiv>
      Complete
    </todoItem__CompleteMarkDiv>
  </todoItem__LowDiv>
</TodoItemDiv>
`;

exports[`TodoItem should render correct with long description 1`] = `
<TodoItemDiv>
  <todoItem__LowDiv>
    <todoItem__ClockDiv>
      <Icon
        type="clock-circle-o"
      />
    </todoItem__ClockDiv>
    <div
      style={
        Object {
          "gridArea": "notification-time",
        }
      }
    >
      1970/01/01 08:00
    </div>
    <todoItem__CreatedTimeDiv>
      <span>
        Creat at: 
      </span>
      1970/01/01 08:00
    </todoItem__CreatedTimeDiv>
    <todoItem__DescriptionDiv>
      <p>
        Hello!
LineTwo
 ...
      </p>
    </todoItem__DescriptionDiv>
    <todoItem__ActionButtonDiv>
      <Connect(TodoActionButton)
        enableComplete={true}
        enableEdit={true}
        todo={
          Object {
            "complete": false,
            "createTime": 0,
            "description": "Hello!
LineTwo
LineThree",
            "dueDatetime": 10,
            "id": "1",
          }
        }
      />
    </todoItem__ActionButtonDiv>
  </todoItem__LowDiv>
</TodoItemDiv>
`;

exports[`TodoItem should render correct with short description 1`] = `
<TodoItemDiv>
  <todoItem__LowDiv>
    <todoItem__ClockDiv>
      <Icon
        type="clock-circle-o"
      />
    </todoItem__ClockDiv>
    <div
      style={
        Object {
          "gridArea": "notification-time",
        }
      }
    >
      1970/01/01 08:00
    </div>
    <todoItem__CreatedTimeDiv>
      <span>
        Creat at: 
      </span>
      1970/01/01 08:00
    </todoItem__CreatedTimeDiv>
    <todoItem__DescriptionDiv>
      <p>
        Hello!
      </p>
    </todoItem__DescriptionDiv>
    <todoItem__ActionButtonDiv>
      <Connect(TodoActionButton)
        enableComplete={true}
        enableEdit={true}
        todo={
          Object {
            "complete": false,
            "createTime": 0,
            "description": "Hello!",
            "dueDatetime": 10,
            "id": "1",
          }
        }
      />
    </todoItem__ActionButtonDiv>
  </todoItem__LowDiv>
</TodoItemDiv>
`;
